js/runner.go

```diff
@@ -753,3 +755,11 @@ func (s *scriptException) StackTrace() string {
 func (s *scriptException) Unwrap() error {
 	return s.inner
 }
+
+func (s *scriptException) Hint() string {
+	return "script exception"
+}
+
+func (s *scriptException) ExitCode() errext.ExitCode {
+	return exitcodes.ScriptException
```

Comment:
Like I mentioned on Slack, it seems out of place to have non-`cmd` packages define what exit codes to use. I would prefer returning `js`-specific errors here, and mapping them to exit codes in `cmd`. If a huge `switch` is unavoidable, I'd still prefer having that in a single place than handling exit codes in internal packages.

Reply:
Hmm I agree this isn't the cleanest solution, but I don't see what we're sacrificing (besides "purity") with this approach, and it seems better than the alternatives :man_shrugging: 

The problem with having a huge `switch`, besides the ugliness of the `switch` itself, is that we'd have to make all of the error types exported, so we can match against them in `cmd`, i.e. everything will actually be much more tightly coupled, even if we ignore the ugly `switch`. In this way, certain local error types would have to know about "exit codes", but it still seems better than the alternative. Besides, with a `switch`, we'd always have to remember to update it every time we add a new error code, while this way errors are localized where they're emitted.

Originally, I thought about having the constants in `errext/exitcodes` be simple `iota` automatically-generated ones, and then to have a huge `map[errext.ExitCode]int` in `cmd/` with the actual `int` exit codes, but that just seems unnecessary and would mean that every Go program that wants to get the error codes will have to import `cmd`.

Reply:
Yeah, the coupling is a problem, though we could just check against known interfaces defined in `cmd` and extract details about the error that would determine the status code it maps to. It's a bit of a roundabout way of doing things, but IMO cleaner than everything importing a global `errext` package.

But I'd have to play around with this to know for sure, so if you think it's unfeasible I'm OK with keeping it as is.

> would mean that every Go program that wants to get the error codes will have to import `cmd`

That makes sense to me since these are CLI (`cmd`) exit codes. If `cmd` is too much to import, would a subpackage like `cmd/errors` be better?

Reply:
btw part of the idea for `errext` was that it would allow us to chip away at some of the `lib/` mess. Making a new global package for just errors might seem a bit much, but we do quite a lot of error handling in k6, so it's not too outlandish.

`errext` (and its sub-packages) wont't import any other k6 package, so I think they can have a somewhat clean-ish structure:
- `/errext/`: interface types and small helpers
- `/errext/exitcodes` - only constants - I can understand why having them in `cmd` makes sense, but having them and nothing else in this package is also very clean (regardless of whether we have a big `switch` in cmd that references them, or whether we stick with my proposal in this PR where every error knows its code)
- `/errext/errorcodes` - this seems like a suitable place for the `error_code` detection, once we take it out of `lib/netext/httpext` so we can use it for other things like web sockets or gRPC